CMakeLists.txt

```diff
@@ -1,142 +1,183 @@
+  GIT_TAG        "develop"
+  PREFIX         "${CMAKE_BINARY_DIR}/g2"
+  CMAKE_ARGS     "-DCMAKE_PREFIX_PATH=${CMAKE_INSTALL_PREFIX}"
+                 "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/g2"
+                 "-DENABLE_TESTS=${ENABLE_TESTS}"
+  )
+
+ExternalProject_Add(g2tmpl
+  GIT_REPOSITORY "https://github.com/noaa-emc/nceplibs-g2tmpl"
+  GIT_TAG        "develop"
+  PREFIX         "${CMAKE_BINARY_DIR}/g2tmpl"
+  CMAKE_ARGS     "-DCMAKE_PREFIX_PATH=${CMAKE_INSTALL_PREFIX}"
+                 "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/g2tmpl"
+                 "-DENABLE_TESTS=${ENABLE_TESTS}"
+  )
+
+ExternalProject_Add(w3emc
+  GIT_REPOSITORY "https://github.com/noaa-emc/nceplibs-w3emc"
+  GIT_TAG        "develop"
+  PREFIX         "${CMAKE_BINARY_DIR}/w3emc"
+  CMAKE_ARGS     "-DCMAKE_PREFIX_PATH=${CMAKE_INSTALL_PREFIX}"
+                 "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/w3emc"
+                 "-DENABLE_TESTS=${ENABLE_TESTS}"
+  DEPENDS        nemsio sigio
+  )
+
+if(BUILD_CRTM)
+  ExternalProject_Add(crtm
+    GIT_REPOSITORY "https://github.com/noaa-emc/emc_crtm"
+    GIT_TAG        "develop"
+    PREFIX         "${CMAKE_BINARY_DIR}/crtm"
+    CMAKE_ARGS     "-DCMAKE_PREFIX_PATH=${CMAKE_INSTALL_PREFIX}"
+                   "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/crtm"
+    )
 endif()
 
-# Flag for using static netCDF
-if(STATIC_IS_DEFAULT)
-  set(NETCDF_USE_STATIC_LIBRARIES TRUE)
-else()
-  set(NETCDF_USE_STATIC_LIBRARIES FALSE)
+if(BUILD_POST)
+  ExternalProject_Add(nceppost
+    GIT_REPOSITORY "https://github.com/aerorahul/emc_post"
+    GIT_TAG        "feature/cmakepkg"
+    PREFIX         "${CMAKE_BINARY_DIR}/nceppost"
+    CMAKE_ARGS     "-DCMAKE_PREFIX_PATH=${CMAKE_INSTALL_PREFIX}"
+                   "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/nceppost"
+                   "-DBUILD_POSTEXEC=OFF"
+    DEPENDS        bacio sigio sfcio gfsio nemsio w3nco w3emc g2 g2tmpl ip sp crtm
+    )
 endif()
-
-add_subdirectory(NCEPLIBS-crtm)
-
-add_subdirectory(NCEPLIBS-bacio)
-add_subdirectory(NCEPLIBS-ip)
-add_subdirectory(NCEPLIBS-g2)
-add_subdirectory(NCEPLIBS-sigio)
-add_subdirectory(NCEPLIBS-sp)
-add_subdirectory(NCEPLIBS-nemsio)
-add_subdirectory(NCEPLIBS-w3nco)
-# add_subdirectory(NCEPLIBS-bufr)
-add_subdirectory(NCEPLIBS-g2tmpl)
-add_subdirectory(NCEPLIBS-gfsio)
-add_subdirectory(NCEPLIBS-landsfcutil)
-add_subdirectory(NCEPLIBS-sfcio)
-add_subdirectory(NCEPLIBS-w3emc)
-add_subdirectory(NCEPLIBS-nemsiogfs)
-
-add_subdirectory(NCEPLIBS-grib_util)
-add_subdirectory(NCEPLIBS-prod_util)
-add_subdirectory(UFS_UTILS)
-add_subdirectory(NCEPLIBS-post)
-
-# Get version from sub-directories
-file(STRINGS "NCEPLIBS-bacio/VERSION" bacioVersion)
-file(STRINGS "NCEPLIBS-nemsio/VERSION" nemsioVersion)
-file(STRINGS "NCEPLIBS-sp/VERSION" spVersion)
-file(STRINGS "NCEPLIBS-w3emc/VERSION" w3emcVersion)
-file(STRINGS "NCEPLIBS-w3nco/VERSION" w3ncoVersion)
-
-# Create shell scripts for bash and csh to set
```

Comment:
We may have to retain this part in the release/public-v1 branch, as well as setenv_nceplibs_bash_template.txt and setenv_nceplibs_csh_template.txt.

Reply:
I am hoping not. If the model build is properly set up, you should'nt need any of this.